File: ProjectRSC/Server/src/org/darkquest/gs/plugins/skills/Prayer.java
package org.darkquest.gs.plugins.skills;

import org.darkquest.gs.model.GameObject;
import org.darkquest.gs.model.Player;
import org.darkquest.gs.plugins.listeners.action.ObjectActionListener;
import org.darkquest.gs.plugins.listeners.executive.ObjectActionExecutiveListener;

public class Prayer implements ObjectActionListener, ObjectActionExecutiveListener {

    @Override
    public void onObjectAction(GameObject object, String command, Player player) {
        if (command.equalsIgnoreCase("recharge at")) {
            player.getActionSender().sendMessage("You recharge at the altar.");
            player.getActionSender().sendSound("recharge");
            int maxPray = object.getID() == 200 ? player.getMaxStat(5) + 2 : player.getMaxStat(5);
            if (player.getCurStat(5) < maxPray) {
                player.setCurStat(5, maxPray);
            }
            player.getActionSender().sendStat(5);
            return;
        }
    }

	@Override
	public boolean blockObjectAction(GameObject obj, String command,
			Player player) {
		if (command.equalsIgnoreCase("recharge at")) {
			return true;
		}
		return false;
	}

}

File: ProjectRSC/Server/scripts/python/skills/Prayer.py
from org.darkquest.gs.plugins.listeners.action import ObjectActionListener
from org.darkquest.gs.plugins.listeners.executive import ObjectActionExecutiveListener
from org.darkquest.gs.plugins import PlugInterface

'''
@author: GORF
Prayer skill
'''
class Prayer(PlugInterface, ObjectActionListener, ObjectActionExecutiveListener):
	
	def onObjectAction(self, gameObject, command, player):
		script = player.getScriptHelper()
		
		script.displayMessage("You recharge at the altar.")
		script.sendSound("recharge")
		maxPray = script.getMaxLevel(player.SkillType.PRAYER)
		cLevel = script.getCurrentLevel(player.SkillType.PRAYER)
		
		if gameObject.getID() == 200:
			maxPray = script.getMaxLevel(player.SkillType.PRAYER) + 2
		if cLevel < maxPray:
			script.restoreStat(player.SkillType.PRAYER, maxPray)
         	
	def blockObjectAction(self, gameObject, command, player):
		return command == "recharge at"
